src/audio/dai.c

```diff
@@ -473,6 +473,14 @@ static int dai_params(struct comp_dev *dev,
 			comp_err(dev, "dai_params(): failed to alloc dma buffer");
 			return -ENOMEM;
 		}
+
+		/*
+		 * dma_buffer frame_fmt should be overwritten with value
+		 * received in sof_ipc_comp_config, because value from params
+		 * represents data format for component connected with dai
+		 */
+		buffer_set_params(dd->dma_buffer, params, true);
```

Comment:
There is dd->dma_buffer->buffer_fmt update. Pls remove if this one is unused (vs. inner audio stream frame fmt).

Reply:
https://github.com/thesofproject/sof/pull/3179